settings/src/components/backup-codes.js

```diff
@@ -71,12 +74,13 @@ function Setup( { setRegenerating } ) {
 	}, [] );
 
 	// Finish the setup process.
-	const handleFinished = useCallback( () => {
-		// The codes have already been saved to usermeta, see `generateCodes()` above.
-		refreshRecord( userRecord ); // This has the intended side-effect of redirecting to the Manage screen.
+	const handleFinished = useCallback( async () => {
 		setGlobalNotice( 'Backup codes have been enabled.' );
 		setRegenerating( false );
-	} );
+		userRecord.record.isSetupFinished = true;
+		// The codes have already been saved to usermeta, see `generateCodes()` above.
+		await refreshRecord( userRecord ); // This has the intended side-effect of redirecting to the Manage screen.
```

Comment:
Not so sure about the meaning of the comment in this line, does it make sense to remove it now?